CONTRIBUTING.md

```diff
@@ -2,6 +2,32 @@
 
 We welcome all contributions!
 
+## Rules of Fn core (ie: what belongs here and what doesn't)
+
+Fn server (core) is only for the essential parts of the entire Fn ecosystem. 
+These include:
+
+- The core API (apps, routes, calls, logs)
+- Executing functions (sync and async)
+- Extension points (callbacks, middleware, API additions)
+
+That's it. Everything else should be built as an extension.
+
+This does not include:
+
+- authentication
+- stats/metrics
+- special/optional features such as triggers, fdk's, workflows, event sources, etc.
+- could be argued that additional I/O formats beyond the basic ones we support should be built as plugins too
+
```

Comment:
Would using the terms plugins or extensions consistently in the documents make things more clear?